addons/pos_online_payment_self_order/static/src/pages/order_cart/order_cart.js

```diff
@@ -0,0 +1,67 @@
+/** @odoo-module */
+import { patch } from "@web/core/utils/patch";
+import { OrderCart } from "@pos_self_order/pages/order_cart/order_cart";
+import { _t } from "@web/core/l10n/translation";
+
+patch(OrderCart.prototype, "pos_online_payment_self_order.OrderCart", {
+    get buttonToShow() {
+        if (this.selfOrder.self_order_mode === "each") {
+            return "Pay";
+        } else if (this.selfOrder.self_order_mode === "meal") {
+            const order = this.selfOrder.currentOrder;
+            if (!order) {
+                return "";
+            }
+            if (!order.isSavedOnServer) {
+                return "Order";
+            } else {
+                return this.selfOrder.has_self_order_online_payment_method ? "Pay" : "Pay at cashier";
+            }
+        } else {
+            return this._super(...arguments);
+        }
+    },
+    async processOrder() {
+        if (this.sendInProgress) {
+            return;
+        }
+
+        if (!this.selfOrder.table) {
+            this.state.selectTable = true;
+            return;
+        }
+
+        if (this.selfOrder.self_order_mode === "meal" && this.selfOrder.has_self_order_online_payment_method) {
+            const order = this.selfOrder.currentOrder;
+            if (!order) {
+                this.selfOrder.notification.add(_t("The current order is invalid."), { type: "danger" });
+                return;
+            }
+            if (!order.isSavedOnServer) {
+                this.sendInProgress = true;
+                await this.selfOrder.sendDraftOrderToServer();
+                this.sendInProgress = false;
+            } else {
+                this.checkAndOpenPaymentPage(order);
+            }
+        } else if (this.selfOrder.self_order_mode === "each") {
+            this.sendInProgress = true;
+            const order = await this.selfOrder.sendDraftOrderToServer();
+            this.sendInProgress = false;
+            this.checkAndOpenPaymentPage(order);
+        } else {
+            return this._super(...arguments);
+        }
+    },
```

Comment:
It should be safer to put the `sendInProgress = false` in the `finally` clause of a `try` statement.

Reply:
Indeed. I didn't check that sendDraftOrderToServer suppresses the error.